torch/_subclasses/functional_tensor.py

```diff
@@ -220,6 +220,12 @@ def __init__(self, pre_dispatch=False, export=False):
         self._mode_key = torch._C._TorchDispatchModeKey.FUNCTIONAL
         # This will be turned off later for pre-dispatch functionalization
         self._dispatch_key = torch._C.DispatchKey.PreDispatch if pre_dispatch else None  # type: ignore[attr-defined]
+        # Map of functions or ScriptObjects to a token. The tokens help keep
+        # track of the ordering between side effectful operations. We have a
+        # token per ScriptObject and a token per function.
+        self._tokens: Dict[
+            Union[torch._ops.OpOverload, torch.ScriptObject], torch.Tensor
+        ] = {}
```

Comment:
It might be overkill, but if we're going down the route of multiple effects, it's more that we should have "effects" and each "effect" should map to a token. Certain groups of ops may share the same effect (e.g. a "printing" effect). So this should be a dict["effect", torch.Tensor].

For starters, we can have one effect ("Ordered"?), and it's probably fine for it to be a string (could also be an enum).